### DISCUSSION TITLE
Github files and folder Icons

### DISCUSSION THREAD

**Discussion by Anu26092002** [community/community#28338] [Other Feature Feedback, Questions, & Ideas] [answered]
It would be a great addition to Github if all the files and folder in a repository had an **icon** associated with it, an unique icon for each file type. 👍🏼 🏴󠁧󠁢󠁷󠁬 ❤️

**Comment by jfullstackdev** [comment]
what icons are those ? 

is this not enough ? 

![image](https://user-images.githubusercontent.com/47092464/182876297-0de32c48-66ce-424f-8809-6eaeb7906c1d.png)

submodule has a unique icon too

**Reply by stretch07** [reply]
No. We want icons like in vscode where you can differentiate file types by their icons. Two or three little icons are not enough!!

**Reply by Anu26092002** [reply]
These are enough but it could be more intuitive like we see in VS Code, so that will also help us to understand what type of file or folder is in the repository.

Something like this
![image](https://user-images.githubusercontent.com/78692178/182878290-673d3e48-7a4f-49b0-bf10-12144586ed39.png)

**Comment by Anu26092002** [comment]
Currently, there are only some icons in Github for folders and files, I personally feel that this feature can be improved by adding unique icons to each file type as used in VS Code.

Some Examples
![image](https://user-images.githubusercontent.com/78692178/182879106-4a44ad7d-b23d-4c55-897b-b299701eecdd.png)
![image](https://user-images.githubusercontent.com/78692178/182879454-e4a45002-957e-4ec8-8325-181ec645057a.png)

**Comment by jfullstackdev** [comment, marked as answer]
@Anu26092002 

this is what you are looking for, the `github.dev` !

simply click the `.` on your keyboard when you are in a repo, or change the URL from `github.com` to `github.dev`

the interface seems familiar, right ? 

![image](https://user-images.githubusercontent.com/47092464/182883048-693313c7-ac8d-48cc-a322-f69373cb3d67.png)

for more info, 

https://docs.github.com/en/codespaces/the-githubdev-web-based-editor

**Reply by Anu26092002** [reply]
Thanks!! Didn't knew about it 👌🏼 🙆🏼‍♂️

**Reply by jfullstackdev** [reply]
the other one is Codespaces, that includes the ability to compile without installing anything, actually, I just finished my Java code using that even without Java, Git and Bash locally, that's the power of Codespaces, and the commands are native to the OS

but anyways, it's a remote Linux machine being given for free by GitHub for personal users, and hopefully it will remain free

you just need to be in the waiting list but it's worth the wait

**Reply by stretch07** [reply]
I knew about github.dev and vscode.dev but it'd be really nice to have these icons at github.com
